### ISSUE TITLE
Building error: relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object

### ISSUE THREAD

**Issue by antage** [meh/rust-hidapi-sys#4]
I added 'build' feature and when I am trying to compile I got following error: 
```
error: linking with `cc` failed: exit code: 1
... [skipped long-long command]
  = note: /usr/bin/ld: /home/********/target/debug/deps/libhidapi_sys-e7ecce37533f01a9.rlib(hid.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC`.
```

OS: Linux Debian 9 amd64
Rust target: x86_64-unknown-linux-gnu

```
$ rust --version
rustc 1.24.0-nightly (f9b0897c5 2017-12-02)
$ gcc --version
gcc (Debian 6.3.0-18) 6.3.0 20170516
```

My Cargo.toml:
```
[dependencies.hid]
version = "^0.4.1"
features = ["build"]
```

I tried to add 'static' feature but it doesn't help.

**Comment by meh** [comment]
Could you run a `cargo update` and try again please?

**Comment by antage** [comment]
hidapi-sys 0.1.4: the same error when I try to link an executable (`cargo test`).

**Comment by meh** [comment]
Try running `cargo run --example list` from the `hid` repository then. Are you building a dylib?

**Comment by antage** [comment]
`cargo run --example list` works fine in hid repo.
But if I add `default = ["build"]` in Cargo.toml's `[features]` section I got the same error.

I work on [cp211x_uart](https://github.com/antage/cp211x_uart). It's not dylib. I got the error when I run `cargo test` and cargo is trying to link a program from `examples` directory.

I see straightforward solution: to add `build.pic(true)` in `build.rs` for linux target. But I'm not sure it is correct.

**Comment by antage** [comment]
I have tried to `build.pic(true)` but it doesn't help.

**Comment by meh** [comment]
Yeah, this is really weird, I honestly have no idea what's going on especially considering it should be trying to link it statically and not dynamically.

You should try asking on IRC if there's anything obvious going on.
